ouroboros-consensus/src/Ouroboros/Consensus/BlockFetchServer.hs

```diff
@@ -76,14 +76,16 @@ blockFetchServer _tracer chainDB registry = senderSide
         registry
         (ChainDB.StreamFromInclusive start)
         (ChainDB.StreamToInclusive   end)
-      return $ case errIt of
+      case errIt of
         -- The range is not in the ChainDB or it forks off more than @k@
         -- blocks back.
-        Left  _  -> SendMsgNoBlocks $ return senderSide
+        Left  _  -> do
+          traceWith tracer (TraceNoBlocks start end)
+          pure $ SendMsgNoBlocks $ return senderSide
         -- When we got an iterator, it will stream at least one block since
         -- its bounds are inclusive, so we don't have to check whether the
         -- iterator is empty.
-        Right it -> SendMsgStartBatch $ sendBlocks it
+        Right it -> pure $ SendMsgStartBatch $ sendBlocks it
```

Comment:
Nitpick: we're using `return` in this module (and in fact, in most of the consensus codebase) instead of `pure`. Just for consistency, stick with `return`, certainly on line 84 which contains both `pure` and `return` :slightly_smiling_face:

Reply:
Done